### PR TITLE
Add Customer Service Tests

### PR DESCRIPTION
**Pull request by scorpionknifes**
Need @prisma/client to construct errors (idk if this is true)

### CODE REVIEW

File: server/core/src/customers/customers.service.spec.ts [hunk 1]
```diff
@@ -33,10 +45,104 @@ describe('CustomersService', () => {
...
+        customerFindOneResponse,
+      );
+      expect(mockPrismaService.customer.findUnique).toBeCalledWith({
+        where: customerFindOneRequest,
+      });
+    });
+    xit('should throw error if customer not found', async () => {
+      // TODO - figure out prisma error handling when no record found
+    });
+  });
```
Comment (scorpionknifes) [inline server/core/src/customers/customers.service.spec.ts:130]: @robertwt7 do you know what the expected behaviour here? thanks

File: server/core/src/customers/testConstants.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { Prisma } from '@prisma/client';
```
Comment (robertwt7) [inline server/core/src/customers/testConstants.ts:1]: `Prisma` already exist in our `database` package and is exported from there. so we shouldn't reinstall prisma client in core

File: server/core/package.json [hunk 3]
```diff
@@ -30,6 +30,7 @@
     "@nestjs/passport": "^9.0.0",
     "@nestjs/platform-express": "^9.0.0",
     "@nestjs/platform-fastify": "^9.2.1",
+    "@prisma/client": "^4.11.0",
```
Comment (robertwt7) [inline server/core/package.json:33]: please revert

File: server/core/src/customers/customers.service.spec.ts [hunk 4]
```diff
@@ -33,10 +45,104 @@ describe('CustomersService', () => {
...
+      );
+      expect(await service.update(customerUpdateRequest)).toStrictEqual(
+        customerUpdateResponse,
+      );
+      expect(mockPrismaService.customer.update).toBeCalledWith(
+        customerUpdateRequest,
+      );
+    });
+    xit('should throw error if customer not found', async () => {
+      // TODO - figure out prisma error handling when no record found
```
Comment (robertwt7) [inline server/core/src/customers/customers.service.spec.ts:145]: similar to above, more code can be found here:
https://www.prisma.io/docs/reference/api-reference/error-reference

**Review by robertwt7 — APPROVED** [review]
lgtm :), great job!

**Comment by scorpionknifes** [comment]
Closes #14

**Pull request merged**